basculehttp/constructor.go

```diff
@@ -24,11 +25,25 @@ const (
 	DefaultHeaderDelimiter = " "
 )
 
+func DefaultGetURLFunc(u *url.URL) (string, error) {
+	return u.EscapedPath(), nil
+}
+
+func CreateRemovePrefixURLFunc(prefix string) func(*url.URL) (string, error) {
```

Comment:
Could the signature of the returned closure not be `func(*url.URL) (*url.URL, error)`?  In other words, return the original URL modified appropriately rather than the `string` representation.  This approach has two (2) benefits:

(1) You or your client can compose things easily, without having to reparse the `string` repeatedly.  For example:
```
type GetURL func(*url.URL) (*url.URL, error)
func removePrefix(prefix) GetURL { /* remove the prefix */ }
func anotherThing(next GetURL) GetURL {
    return func(u *url.URL) (*url.URL, error) {
        /* do some other stuff */
        return next(u)
    }
}
```

(2) A `*url.URL` can be useful to a client who wants to do further processing outside `bascule`.  This is related to (1), but has more to do with a client using the results of a `GetURL` strategy rather than your code.

Since `*url.URL` is mutable, it's generally good practice to make a copy of the HTTP request's `URL` member prior to invoking `GetURL`.  You can do this very simply with `u := *request.URL`, in which case `u` is of type `url.URL` and you'll pass `&u` around.